### ISSUE TITLE
[QUESTION] MoE score precision

### ISSUE THREAD

**Issue by Daniel5103** [NVIDIA/Megatron-LM#2741]
In megatron/core/transformer/moe/moe_utils.py
 
compute_routing_scores_for_aux_loss
if score_function == "softmax":
        scores = torch.softmax(logits, dim=-1, dtype=torch.float32)
    elif score_function == "sigmoid":
        scores = torch.sigmoid(logits)
        scores = scores / (scores.sum(dim=-1, keepdim=True) + 1e-20)
    else:
        raise ValueError(f"Invalid score_function: {score_function}")

Why softmax in FP32, but sigmoid in original dtype?
Should it be changed to the same high precision?

**Comment by CodersAcademy006** [comment]
Softmax explicitly requires `float32` because `exp` sums easily overflow or lose precision in BF16/FP16. While `sigmoid` is bounded [0, 1], the subsequent normalization step (`scores / scores.sum()`) can still suffer from accumulation errors in lower precision, especially as the number of experts increases.

For auxiliary loss stability, it is safer to compute both in `float32`. A PR to cast `logits` to float32 in the sigmoid branch would be a welcome fix.
